trackbase/FairRKPropagator.cxx

```diff
@@ -0,0 +1,625 @@
+  }
+}
+//______________________________________________________________________________
+void FairRKPropagator::Propagate(double Charge, double* vecRKIn, double* Pos)
+{
+  double diff = Pos[2] - vecRKIn[2];
+  fMaxStep = diff/25;
+  double res_old= diff;
+  double res = 100.0;
+  double vecRKOut[7];
+  double vecRKOutT[7];
+  for (int i=0; i< 7; i++) {vecRKOut[i]=0; vecRKOutT[i]=0;}
+
+  int nIter=0;
+  do {
+    Step(Charge,vecRKIn,vecRKOut);
+    res=(vecRKOut[2]-Pos[2])/diff;
+    if( TMath::Abs(res)< 0.01 || res >res_old ) {
+      break;
+    } else {
+      for (int i=0; i< 7; i++) {
+        vecRKOutT[i]=vecRKOut[i];
+        vecRKIn[i]=vecRKOut[i];
+      }
+    }
+    if(nIter++>1000) { break; }
+  } while(1);
+  if (res > res_old) for (int k=0; k< 7; k++) { vecRKOut[k]=vecRKOutT[k]; }
+  for (int k=0; k< 3; k++) { printf(" vecRKOut[%i] =%f ", k, vecRKOut[k] ); }
+  printf("\n");
+}
+//______________________________________________________________________________
+double FairRKPropagator::Step(double Charge, double* vecRKIn, double* vecOut)
+{
+
+  double vecRKOut[7];
+  for (int i=0; i< 7; i++) { vecRKOut[i]=0; }
+//  for (int i=0; i< 7; i++) printf( "vectRKIn(%i)=%f  \n",i ,vecRKIn[i]);
+//  printf(" ---------------------------------------------------------------- \n");
+  double stepLength = OneStepRungeKutta(Charge,fMaxStep, vecRKIn, vecRKOut);
+  //  printf(" now at     x=%f  y=%f  z=%f \n", vecRKOut[0],vecRKOut[1],vecRKOut[2]);
+  vecOut[0] = vecRKOut[0];
+  vecOut[1] = vecRKOut[1];
+  vecOut[2] = vecRKOut[2];
+  vecOut[6] = vecRKOut[6];
+  vecOut[3] = vecRKOut[3];
+  vecOut[4] = vecRKOut[4];
+  vecOut[5] = vecRKOut[5];
+
+  return stepLength;
+}
+
+//______________________________________________________________________________
+double FairRKPropagator::OneStepRungeKutta(double charge, double step,
+    double* vect, double* vout)
+{
+
+  // Wrapper to step with method RungeKutta.
+
+  /// ******************************************************************
+  /// *                *
+  /// *  Runge-Kutta method for tracking a particle through a magnetic *
+  /// *  field. Uses Nystroem algorithm (See Handbook Nat. Bur. of   *
+  /// *  Standards, procedure 25.5.20)         *
+  /// *                *
+  /// *  Input parameters            *
+  /// * CHARGE    Particle charge        *
+  /// * STEP    Step size          *
+  /// * VECT    Initial co-ords,direction cosines,momentum   *
+  /// *  Output parameters             *
+  /// * VOUT    Output co-ords,direction cosines,momentum  *
+  /// *  User routine called             *
+  /// * CALL GUFLD(X,F)            *
+  /// *                *
+  /// *    ==>Called by : <USER>, GUSWIM         *
+  /// * Authors    R.Brun, M.Hansroul  *********     *
+  /// *      V.Perevoztchikov (CUT STEP implementation)  *
+  /// *                *
+  /// *                *
+  /// ******************************************************************
+
+  double h2, h4, f[4];
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'h4' can be reduced.](https://app.codacy.com/app/FairRootGroup/FairRoot/pullRequest?prid=4139720)